docs/operations/karpenter.md

```diff
@@ -2,68 +2,117 @@
+
 ## Installing
 
-If using kOps 1.26 or older, enable the Karpenter feature flag :
+### New clusters
 
 ```sh
-export KOPS_FEATURE_FLAGS="Karpenter"
+export NAME="my-cluster.example.com"
+export REGION="us-east-1"
+export ZONE="us-east-1a"
+
+kops create cluster --name ${NAME} \
+  --state=s3://my-state-store \
+  --discovery-store=s3://my-discovery-store \
+  --cloud=aws \
+  --networking=cilium \
+  --zones=${ZONE} \
+  --instance-manager=karpenter \
+  --yes
 ```
 
-Karpenter requires that external permissions for ServiceAccounts be enabled for the cluster. See [AWS IAM roles for ServiceAccounts documentation](/cluster_spec#service-account-issuer-discovery-and-aws-iam-roles-for-service-accounts-irsa) for how to enable this. 
-
 ### Existing clusters
 
-On existing clusters, you can create a Karpenter InstanceGroup by adding the following to its InstanceGroup spec:
+The Karpenter addon must be enabled in the cluster spec:
 
 ```yaml
 spec:
-  manager: Karpenter
+  karpenter:
+    enabled: true
 ```
 
-You also need to enable the Karpenter addon in the cluster spec:
+To create a Karpenter InstanceGroup, set the following in its InstanceGroup spec:
 
 ```yaml
 spec:
-  karpenter:
-    enabled: true
+  manager: Karpenter
 ```
 
-### New clusters
-
-On new clusters, you can simply add the `--instance-manager=karpenter` flag:
+### EC2NodeClass and NodePool
 
 ```sh
-kops create cluster --name mycluster.example.com --cloud aws --networking=amazonvpc --zones=eu-central-1a,eu-central-1b --master-count=3 --yes --discovery-store=s3://discovery-store/
+export USER_DATA=$(aws s3 cp s3://my-state-store/${NAME}/igconfig/node/nodes/nodeupscript.sh -)
+
+cat <<EOF | kubectl apply -f -
+apiVersion: karpenter.k8s.aws/v1
+kind: EC2NodeClass
+metadata:
+  name: default
+spec:
+  amiFamily: Custom
+  amiSelectorTerms:
+    - ssmParameter: /aws/service/canonical/ubuntu/server/24.04/stable/current/amd64/hvm/ebs-gp3/ami-id 
+  associatePublicIPAddress: true
+  tags:
+    KubernetesCluster: ${NAME}
+    kops.k8s.io/instancegroup: nodes
+    k8s.io/role/node: "1"
+  subnetSelectorTerms:
+    - tags:
+        KubernetesCluster: ${NAME}
+  securityGroupSelectorTerms:
+    - tags:
+        KubernetesCluster: ${NAME}
+        Name: nodes.${NAME}
+  instanceProfile: nodes.${NAME}
+  userData: |
+$(echo "$USER_DATA" | sed 's/^/    /')
+EOF
+
+cat <<EOF | kubectl apply -f -
+apiVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  name: default
+spec:
+  template:
+    spec:
+      requirements:
+        - key: kubernetes.io/arch
+          operator: In
+          values: ["amd64", "arm64"]
+        - key: kubernetes.io/os
+          operator: In
+          values: ["linux"]
+        - key: karpenter.sh/capacity-type
+          operator: In
+          values: ["on-demand", "spot"]
+      nodeClassRef:
+        group: karpenter.k8s.aws
+        kind: EC2NodeClass
+        name: default
+      expireAfter: 24h
```

Comment:
We can add comments here right - I think this means that VMs will not last longer than 24 hours?  That is probably worth adding a comment, example yaml has a habit of being copied and pasted around the internet and I can imagine everyone being surprised that their VMs are shutting down every day (assuming it means what I think it means!)